packages/calypso-sentry/src/index.ts

```diff
@@ -118,36 +118,66 @@ function beforeBreadcrumb( breadcrumb: SentryApi.Breadcrumb ): SentryApi.Breadcr
 	return breadcrumb;
 }
 
-function shouldEnableSentry(): boolean {
+function shouldEnableSentry( sampleRate: number ): boolean {
 	// This flag overrides the other settings, always enabling Sentry:
 	if ( config.isEnabled( 'always-enable-sentry' ) ) {
 		return true;
 	}
 
-	// Only load for 10% of requests when Sentry is enabled in the environment:
-	return config.isEnabled( 'catch-js-errors' ) && Math.floor( Math.random() * 10 ) === 1;
+	// By default, only load for 10% of requests when Sentry is enabled in the environment:
+	return (
+		config.isEnabled( 'catch-js-errors' ) &&
+		Math.ceil( Math.random() * 10 ) <= Math.floor( sampleRate * 10 )
+	);
 }
 
 interface SentryOptions {
-	beforeSend: ( e: SentryApi.Event ) => SentryApi.Event | null;
+	beforeSend?: ( e: SentryApi.Event ) => SentryApi.Event | null;
 	userId?: number;
+	sampleRate?: number;
 }
-export async function initSentry( { beforeSend, userId }: SentryOptions ) {
+
+/**
+ * Sentry initialization function.
+ * It can be called multiple times, but it will only initialize Sentry once, if the sampleRate is met (default 10%).
+ * It stores the previous execution parameters, and will use them as default if the function is called again.
+ * @param parameters Initialization parameters
+ * @returns A promise that resolves to the previous state of Sentry (enabled/disabled)
+ */
+export async function initSentry( parameters?: SentryOptions ) {
 	// Make sure we don't throw
 	try {
 		// No Sentry loading on the server.
+		if ( typeof document === 'undefined' ) {
+			return 'skipped'; // Previous state: skipped
+		}
+
 		// No double-loading.
-		if ( typeof document === 'undefined' || state.state !== 'initial' ) {
-			return;
+		// No initializing with empty params
+		if (
+			( state.state === 'initial' && ! parameters ) ||
+			! [ 'initial', 'disabled' ].includes( state.state )
+		) {
+			return 'enabled'; // Previous state: enabled -> We're already running [loading, loaded, error]
 		}
-		state = { state: 'loading' };
+
+		// We use previous invocation parameters here, allowing the user to override any of them
+		// We need this because some of the params are initialized at app boot, and can't be recreated when the app is already running
+		const params =
+			state.state === 'initial'
+				? ( parameters as SentryOptions ) // We know it's not undefined because of the check above
+				: { ...state.params, sampleRate: 0.1, ...parameters };
```

Comment:
We don't need to store a default sampleRate in state.params since we set a default value. IMO, we can put the default `sampleRate` value into the `shouldEnableSentry` function so that it's closer to where it's used